### PR TITLE
Fix hints for Blue Fire Arrows

### PR DESCRIPTION
**Pull request by mracsys**
Adds a distinct Blue Fire Arrows item to the pool for hint and logic purposes. A unit test was created to test if the item shows up as way of the hero when all bottles are locked behind red ice, and Light Arrows are locked behind the Ice Cavern song.

### CODE REVIEW

File: Patches.py [hunk 1]
```diff
@@ -2290,14 +2290,6 @@ def update_scrub_text(message, text_replacement, default_price, price, item_name
...
         rom.write_byte(0xCA61E3, torch_count)
 
-    if world.settings.blue_fire_arrows:
-        rom.write_byte(0xC230C1, 0x29) #Adds AT_TYPE_OTHER to arrows to allow collision with red ice
-        rom.write_byte(0xDB38FE, 0xEF) #disables ice arrow collision on secondary cylinder for red ice crystals
-        rom.write_byte(0xC9F036, 0x10) #enable ice arrow collision on mud walls
-        #increase cylinder radius/height for red ice sheets
-        rom.write_byte(0xDB391B, 0x50)
-        rom.write_byte(0xDB3927, 0x5A)
-
```
Comment (Cuphat) [inline Patches.py:2300]: Is there a reason this was removed?

Comment (mracsys) [reply Patches.py:2300]: It got duplicated at some point. It's patched in a few lines above. I tested red ice and mud walls to be sure, and it doesn't look like anything is touching those bytes in between.

Comment (Cuphat) [reply Patches.py:2300]: Answered in Discord. It's duplication.

**Review by Cuphat — APPROVED** [review]

**Comment by r0bd0g** [comment]
Do you still need to be checking the blue fire arrows setting in the blue fire helper?

**Comment by fenhl** [comment]
No, that's now redundant.

**Comment by fenhl** [comment]
This PR has unintuitive consequences for plando authors. Would it be possible to make the two items separate in the C code as well, have the get item function for BFAs give ice arrows and set the flag that was previously set by the setting, and have the setting only change the item pool? That way, plandos could have both items in them at the same time and have them work as expected.

**Comment by mracsys** [comment]
I'm not sure what flag you are referring to.

Plando'd Ice Arrows still work if Blue Fire Arrows are on. The only change is they will not be considered logically, only Blue Fire Arrows.

Plando with Blue Fire Arrows and Ice Arrows in a shop, setting is on.
image

**Comment by mracsys** [comment]
> No, that's now redundant.

The setting is still relevant in logic because the Blue Fire Arrows item will not melt red ice if the setting is disabled. They can be introduced into the item pool by plando with the setting off. That should probably be changed.

**Comment by fenhl** [comment]
What I'm saying is finding the Blue Fire Arrows item should always enable the added behavior, and Ice Arrows should not.

**Comment by mracsys** [comment]
Making it dynamic would take a bit. Right now it's controlled by collision hacks at patch time. The pause menu text also overwrites the original Ice Arrow text. I can change the melting behavior, but I don't know how to make the item text update.

**Comment by Cuphat** [comment]
I think making them work as separate items in C is a good goal eventually; for this PR specifically though, it might need some plando check to either replace Ice Arrows/Blue Fire Arrows with the appropriate one for the setting, error out, or something else.

**Pull request merged**